sdks/java/core/src/main/java/org/apache/beam/sdk/metrics/CounterCell.java

```diff
@@ -41,44 +41,32 @@
    */
   CounterCell() {}
 
-  /** Increment the counter by the given amount. */
-  private void add(long n) {
+  /**
+   * Increment the counter by the given amount.
+   * @param n value to increment by. Can be negative to decrement.
+   */
+  public void update(long n) {
     value.addAndGet(n);
     dirty.afterModification();
   }
 
   @Override
-  public DirtyState getDirty() {
-    return dirty;
-  }
-
-  @Override
-  public Long getCumulative() {
-    return value.get();
-  }
-
-  @Override
-  public Counter getInterface() {
-    return this;
-  }
-
-  @Override
-  public void inc() {
-    add(1);
+  public void update(Long n) {
```

Comment:
Could also solve this using the approach taken in Sum Long CombineFn. Specifically, instead of `MetricCell<Counter, Long>` make this `MetricCell<Counter, long[]>`, where the array is expected to be one element. This avoids the boxing/unboxing.

Alternative, the boxing/unboxing here may be OK as long as we aren't using a `Long` as the accumulator/value holder.